src/store/ProductsStore/ProductsStore.ts

```diff
@@ -0,0 +1,116 @@
+import { API_ENDPOINTS } from "@configs/api";
+import { Product } from "@pages/ProductDetail";
+import rootStore from "@store/RootStore/instance";
+import { Meta } from "@utils/meta";
+import { ILocalStore } from "@utils/useLocalStore";
+import axios from "axios";
+import {
+  action,
+  computed,
+  IReactionDisposer,
+  makeObservable,
+  observable,
+  reaction,
+  runInAction,
+} from "mobx";
+
+type searchTitle = undefined | string | string[] | qs.ParsedQs | qs.ParsedQs[];
+
+type PrivateFields =
+  | "_totalProductsList"
+  | "_meta"
+  | "_searchTitle"
+  | "_productsPage"
+  | "_hasMoreData";
+
+export default class ProductsStore implements ILocalStore {
+  private _totalProductsList: Product[] = [];
+  private _meta: Meta = Meta.initial;
+  private _searchTitle: searchTitle = "";
+  private _productsPage: number = 0;
+  private _hasMoreData: boolean = true;
+
+  constructor() {
+    makeObservable<ProductsStore, PrivateFields>(this, {
+      _totalProductsList: observable.ref,
+      _meta: observable,
+      _searchTitle: observable,
+      _productsPage: observable,
+      _hasMoreData: observable,
+      totalProductsList: computed,
+      meta: computed,
+      searchTitle: computed,
+      hasMoreData: computed,
+      productsPage: computed,
+      getProductsList: action,
+      clearProductList: action,
+    });
+  }
+
+  get totalProductsList(): Product[] {
+    return this._totalProductsList;
+  }
+
+  get meta(): Meta {
+    return this._meta;
+  }
+  get searchTitle(): searchTitle {
+    return this._searchTitle;
+  }
+  set searchTitle(value: searchTitle) {
+    this._searchTitle = value;
+  }
```

Comment:
пустые строки, пожалуйста. в целом, если тебе подходит, можешь попробовать подрубить правило линтера lines-between-class-members: https://eslint.org/docs/latest/rules/lines-between-class-members. ну или можешь, если хочешь, другое что-то поискать по теме автоматического линтинга таких штук
```suggestion
  get meta(): Meta {
    return this._meta;
  }
  
  get searchTitle(): searchTitle {
    return this._searchTitle;
  }
  
  set searchTitle(value: searchTitle) {
    this._searchTitle = value;
  }
```